src/sdk/bucketspace_multi_fs.js

```diff
@@ -0,0 +1,399 @@
+                throw new RpcError('MISSING_NSFS_ACCOUNT_CONFIGURATION');
+            }
+            this.validate_bucket_creation(params);
+            const { name } = params;
+            const bucket_path = path.join(this.fs_root, name);
+            console.log(`BucketSpaceFS: create_bucket ${bucket_path}`);
+            // eslint-disable-next-line no-bitwise
+            const unmask_mode = config.BASE_MODE_DIR & ~config.NSFS_UMASK;
+            await nb_native().fs.mkdir(this.fs_context, bucket_path, unmask_mode);
+
+            // bucket configuration file start
+            console.log('creating bucket on default namespace resource', sdk.requesting_account.nsfs_account_config);
+            if (!sdk.requesting_account.nsfs_account_config || !sdk.requesting_account.nsfs_account_config.new_buckets_path) {
+                throw new RpcError('MISSING_NSFS_ACCOUNT_CONFIGURATION');
+            }
+            const bucket = this.new_bucket_defaults(params.name, sdk.requesting_account.email,
+                sdk.requesting_account._id, params.tag, params.lock_enabled);
+            bucket.path = path.join(
+                sdk.nsfs_fs_root, sdk.requesting_account.nsfs_account_config.new_buckets_path, params.name);
+            bucket.should_create_underlying_storage = true;
+            bucket.force_md5_etag = params.force_md5_etag;
+            const create_bucket = JSON.stringify(bucket);
+            await nb_native().fs.writeFile(
+                this.fs_context,
+                bucket_path + ".json",
+                Buffer.from(create_bucket), {
+                    mode: this.get_umasked_mode(config.BASE_MODE_FILE)
+                }
+            );
+        } catch (err) {
+            if (err.code === 'ENOENT') {
+                console.error('BucketSpaceFS: root dir not found', err);
+                throw new S3Error(S3Error.NoSuchBucket);
+            }
+            throw err;
+        }
+    }
+
+    get_umasked_mode(mode) {
+        // eslint-disable-next-line no-bitwise
+        return mode & ~config.NSFS_UMASK;
+    }
+
+
+    new_bucket_defaults(name, email, owner_account_id, tag, lock_enabled) {
+        return {
+            name: name,
+            tag: js_utils.default_value(tag, ''),
+            owner_account: owner_account_id,
+            system_owner: new SensitiveString(email),
+            bucket_owner: new SensitiveString(email),
+            versioning: config.WORM_ENABLED && lock_enabled ? 'ENABLED' : 'DISABLED',
+            object_lock_configuration: config.WORM_ENABLED ? {
+                object_lock_enabled: lock_enabled ? 'Enabled' : 'Disabled',
+            } : undefined,
+        };
+    }
+
+    async delete_bucket(params) {
+        try {
+            const { name } = params;
+            const bucket_path = path.join(this.fs_root, name);
+            console.log(`BucketSpaceFS: delete_fs_bucket ${bucket_path}`);
+            await nb_native().fs.rmdir(this.fs_context, bucket_path);
```

Comment:
// TODO we want to have the logic from bucketspace_nb to separate between two cases:
// - either we should_create_underlying_storage (uls) and then we delete the bucket itself
// - or we just remove the bucket.json and keep the bucket files as is.